src/ApplicationInsights.ServiceFabric.Native/Net45/CorrelatingServiceRemotingClient.cs

```diff
@@ -0,0 +1,177 @@
+    using Microsoft.ServiceFabric.Services.Remoting;
+    using Microsoft.ServiceFabric.Services.Remoting.Client;
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Fabric;
+    using System.IO;
+    using System.Linq;
+    using System.Runtime.Serialization;
+    using System.Threading.Tasks;
+    using System.Xml;
+
+    /// <summary>
+    /// Service remoting client that wraps another service remoting client and adds correlation id and context propagation support. This allows
+    /// traces the client and the service to log traces with the same relevant correlation id and context.
+    /// </summary>
+    public class CorrelatingServiceRemotingClient : IServiceRemotingClient, IWrappingClient
+    {
+        private Uri serviceUri;
+        private Lazy<DataContractSerializer> baggageSerializer;
+        private TelemetryClient telemetryClient;
+        private ServiceContext serviceContext;
+        private ITelemetryInitializer fabricTelemetryInitializer;
+
+        /// <summary>
+        /// Initializes the <see cref="CorrelatingServiceRemotingClient"/> object. It wraps the given inner client object for all the core
+        /// remote call operation.
+        /// </summary>
+        /// <param name="innerClient">The client object which this client wraps.</param>
+        /// <param name="serviceUri">The target Uri of the service which this client will call.</param>
+        public CorrelatingServiceRemotingClient(IServiceRemotingClient innerClient, Uri serviceUri)
+            : this(innerClient, serviceUri, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes the <see cref="CorrelatingServiceRemotingClient"/> object. It wraps the given inner client object for all the core
+        /// remote call operation. It allows passing in a ServiceContext object associated with the caller so telemetry can be logged with
+        /// service fabric properties.
+        /// </summary>
+        /// <param name="innerClient">The client object which this client wraps.</param>
+        /// <param name="serviceUri">The target Uri of the service which this client will call.</param>
+        /// <param name="clientServiceContext">The service context of the caller creating this client object. Pass in null if there isn't one.</param>
+        public CorrelatingServiceRemotingClient(IServiceRemotingClient innerClient, Uri serviceUri, ServiceContext clientServiceContext)
+        {
+            if (innerClient == null)
+            {
+                throw new ArgumentNullException(nameof(innerClient));
+            }
+            if (serviceUri == null)
+            {
+                throw new ArgumentNullException(nameof(serviceUri));
+            }
+
+            this.serviceContext = clientServiceContext;
+            this.fabricTelemetryInitializer = FabricTelemetryInitializerExtension.CreateFabricTelemetryInitializer(this.serviceContext);
```

Comment:
I think we need to address the overall ease of use for AI.SF SDK. In the case of ASP.NET core, getting users to add another line to initialize the telemetry means we are farther away from the goal of zero work to enable it.

Having said that, I agree with you that it is not logically consistent and correct as it is right. I was torn between this and trying to make this more automatic. I will go with your suggestions, and later we should revisit for an overall story of how to make AI.SF easier to configure.